### DISCUSSION TITLE
how to import a const into a useState function

### DISCUSSION THREAD

**Discussion by catakul** [neuefische/questions#39] [Web] [answered]
Ich hab das Periodensystem mit css grid nachgebaut.
Die darin enthaltenen Informationen (Element Symbol, Nummer und Name) stammen aus einem Array in meiner JSON Datei, die ich mit Hilfe der map Funktion anzeigen lasse.
Dieses Array beinhaltet alle Informationen zu jedem einzelnen Element, verpackt in Objekten.
Der nächste Schritt wäre es mir diese Objekte in einem popup-Fenster anzeigen zu lassen.
Hierfür muss ich meine Konstante in einem useState packen um anschließend über React Modals die popup Fenster erstellen zu können.
Bevor ich mir den Kopf zerbreche wie Modals funktionieren, ist die Frage wie ich das useState verwende (siehe letzten Screenshot, Zeile 2 welche aktuell auskommentiert ist)

![Bildschirmfoto 2022-08-02 um 14 10 52](https://user-images.githubusercontent.com/106112634/182372109-cbbd8bd1-18c6-4782-8a23-f95b1ddd6691.png)
![Bildschirmfoto 2022-08-02 um 14 04 51](https://user-images.githubusercontent.com/106112634/182372120-10cb7baa-b481-4b3e-aa88-cd577291cb61.png)
![Bildschirmfoto 2022-08-02 um 14 04 41](https://user-images.githubusercontent.com/106112634/182372129-cc5f2dcf-a6e0-4407-9513-0664c5bef050.png)

**Comment by pixelass** [comment, marked as answer]
Versuch mal ausserhalb (auf oberster Ebene in der Komponente, das `useState` zu verwenden.

Anbei eine reduzierte Form deines Ansatzes.

```jsx
function MyComponent() {
  const [info, setInfo] = useState(null);
  function handleClick(elementNumber) { // besser wäre es hier eine ID zu verwenden.
    // Zuerst finden wir das Element anhand der Nummer
    const element = elements.find(element_ => element_.number === elementNumber);
    // Danach wird es in den State gesendet
    setInfo(element); 
  }

  return <div>{element?.name}</div>
}
```


> Tipp  
> Es wäre leichter Code als Codeblock (nicht als Image) hier zu posten, dadurch tun wir uns leichter eine Lösung vorzuschlagen.
> Am besten ein Link zum aktuellen Code oder noch besser das Problem isolieren und in einer Codesandbox bereitstellen. Wir haben dazu einen kleinen Guide geschrieben: https://github.com/neuefische/questions/wiki/How-to-ask-good-questions
